base/src/main/scala/data/ProCompose.scala

```diff
@@ -0,0 +1,54 @@
+package scalaz
+package data 
+
+object ProCompose {
+  import typeclass.{Profunctor, Strong, Category, StrongClass, CategoryClass}
+  import typeclass.Strong._
+  import typeclass.Profunctor._
+
+  trait ProCompose[P[_, _], A, B] { } 
+
+  case class Pure[P[_, _], A, B](pab: A => B) extends ProCompose[P, A, B]
+
+  case class Composed[P[_, _], A, X, B](pab: P[A, X], more: ProCompose[P, X, B]) extends ProCompose[P, A, B]
```

Comment:
Sorry for being unspecific; when I say "it", I mean an implementation of this `ProCompose` which has a stack-safe `foldMap` function, for *all* `F[_, _]` with a `Semiarrow` instance. It *is* possible to write a stack-safe `foldMap` for a given  `ProCompose`, *but* it requires writing specific code for every `F[_, _]` and is thus impossible to abstract over. That's the problem `Evaluable` solves.